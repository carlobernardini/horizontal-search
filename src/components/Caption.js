import React from 'react';
import PropTypes from 'prop-types';
import { DISPLAY_NAME_CAPTION } from '../constants';

const Caption = ({ children }) => {
    if (!children) {
        return null;
    }

    return (
        <div
            className="orizzonte__filter-caption"
        >
            { children }
        </div>
    );
};

Caption.displayName = DISPLAY_NAME_CAPTION;

Caption.propTypes = {
    children: PropTypes.string
};

Caption.defaultProps = {
    children: null
};

export default Caption;
