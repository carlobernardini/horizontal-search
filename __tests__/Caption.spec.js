import React from 'react';
import Caption from '../src/components/Caption';

describe('<Caption />', () => {
    it('should render a filter caption', () => {
        const wrapper = shallow(
            <Caption>
                This is a filter caption
            </Caption>
        );

        expect(wrapper).toMatchSnapshot();
    });
    it('should not render an empty caption', () => {
        const wrapper = shallow(
            <Caption />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
