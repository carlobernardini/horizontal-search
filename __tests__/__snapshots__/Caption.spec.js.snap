// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Caption /> should not render an empty caption 1`] = `""`;

exports[`<Caption /> should render a filter caption 1`] = `
<div
  className="orizzonte__filter-caption"
>
  This is a filter caption
</div>
`;
